### PR TITLE
Build per-user watched video tracking system

### PR DESCRIPTION
**Pull request by Darkflib**
Implement comprehensive watched videos feature with backend tracking, API endpoints, and UI filtering capabilities.

Backend Changes:
- Add WatchedVideo model with user_id, video_id, channel_id
- Create database migration for watched_videos table
- Add CRUD operations: mark_video_watched, unmark_video_watched, get_watched_video_ids
- Create /api/watched router with POST, DELETE, GET endpoints
- Enhance feed endpoint to include watched status for each video
- Add comprehensive test suite (20 tests, all passing)

Frontend Changes:
- Update API client with watched video methods
- Add WatchedFilter component with eye icon toggle
- Enhance VideoCard with dimmed styling for watched videos
- Add manual mark/unmark buttons with loading states
- Auto-mark videos as watched when clicking YouTube links
- Implement hide/show watched videos filtering
- Display watched counts in filter toggle

Features:
- Per-user watched lists with cascade delete
- Optimistic UI updates for mark/unmark operations
- Visual indicators (checkmark badge, dimmed opacity)
- Filter toggle to hide watched videos from feed
- Unique constraint prevents duplicate entries
- Rate limiting on watched endpoints

Tests: 20 new backend tests covering CRUD, API, feed integration, edge cases, and multi-user isolation

Documentation:
- Update README with watched videos feature
- Mark v2 roadmap item as complete

## Summary
- 

## Checklist
- [x] Tests added/updated
- [x] Docs updated (README/CHANGELOG)
- [x] Affected code has owners’ review
- [x] Code is licensed as required in the CONTRIBUTING.md file

## Security
- Notes on threat model / permissions / secrets

## Release Notes

* **New Features**
  * Mark and unmark videos as watched
  * Toggle filter to hide or show watched videos in your feed
  * Watched status now displayed on video cards

* **Documentation**
  * Updated feature documentation and project roadmap

* **Tests**
  * Added comprehensive test coverage for watched video functionality

### CODE REVIEW

File: app/api/routes_watched.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+router = APIRouter(prefix="/api/watched", tags=["watched"])
+limiter = Limiter(key_func=get_remote_address)
+
+
+class MarkWatchedRequest(BaseModel):
+    """Request model for marking a video as watched."""
+
+    video_id: str
+    channel_id: str
```
[the remark this answers is not in the document]
Comment (Darkflib) [reply app/api/routes_watched.py]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/Darkflib/yt-feed-aggregator/pull/29#discussion_r2501555764)

File: app/db/models.py [hunk 2]
```diff
@@ -55,3 +55,28 @@ class UserChannel(Base):
...
+
+class WatchedVideo(Base):
+    """User's watched videos tracking."""
+
+    __tablename__ = "watched_videos"
+    __table_args__ = (
+        # Ensure a user can only mark a video as watched once
+        # Using index with unique constraint for efficient lookups
+        {"sqlite_autoincrement": False},
+    )
```
[the remark this answers is not in the document]
Comment (Darkflib) [reply app/db/models.py]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/Darkflib/yt-feed-aggregator/pull/29#discussion_r2501555786)

File: app/api/routes_watched.py [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+        Watched video details including video_id, channel_id, and watched_at timestamp
+
+    Raises:
+        HTTPException: 400 if video_id or channel_id is empty
+    """
+    # Validate input
+    if not body.video_id or not body.video_id.strip():
+        raise HTTPException(status_code=400, detail="video_id cannot be empty")
+    if not body.channel_id or not body.channel_id.strip():
+        raise HTTPException(status_code=400, detail="channel_id cannot be empty")
```
[the remark this answers is not in the document]
Comment (Darkflib) [reply app/api/routes_watched.py]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/Darkflib/yt-feed-aggregator/pull/29#discussion_r2501555797)

File: frontend/src/api/client.ts [hunk 4]
```diff
@@ -146,6 +147,32 @@ class APIClient {
...
+  }
+
+  /**
+   * Unmark a video as watched
+   */
+  async unmarkVideoWatched(video_id: string): Promise<void> {
+    await this.request<void>(`/api/watched/${video_id}`, {
+      method: 'DELETE',
+    });
+  }
```
[the remark this answers is not in the document]
Comment (Darkflib) [reply frontend/src/api/client.ts]: @copilot In frontend/src/api/client.ts around lines 164 to 168, the unmarkVideoWatched
request assumes the response has JSON but the DELETE endpoint returns 204 No
Content causing response.json() to throw; update the request handling so it does
not call response.json() for empty responses — specifically detect 204 responses
or an empty body (e.g., response.status === 204 or no
Content-Type/Content-Length) and return void instead of attempting to parse
JSON, while still parsing JSON for non-empty responses and propagating errors as
before.
